src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -629,6 +633,111 @@ theorem shiftLeft_shiftLeft {w : Nat} (x : BitVec w) (n m : Nat) :
     getLsb (x >>> i) j = getLsb x (i+j) := by
   unfold getLsb ; simp
 
+/-! ### sshiftRight -/
+
+theorem sshiftRight_eq {x : BitVec n} {i : Nat} :
+    x.sshiftRight i = BitVec.ofInt n (x.toInt >>> i) := by
+  apply BitVec.eq_of_toInt_eq
+  simp [BitVec.sshiftRight]
+
+/-- The MSB of a bitvector is `true` iff its value interpreted as a natural number in 2s complement
+  is larger than 2^(half the bitwidth). -/
+theorem msb_eq_true_iff_large {x : BitVec w}
+    : x.msb = true ↔ 2 * x.toNat ≥ 2^w  := by
+  rcases w with rfl | w <;> try simp <;> try omega
+  constructor
+  · intro h
+    rw [BitVec.msb_eq_decide] at h
+    simp at h
+    omega
+  · intro h
+    rw [BitVec.msb_eq_decide]
+    simp
+    omega
+
+/-- The MSB of a bitvector is `false` iff its value is smaller than 2^(half the bitwidth).
+  This is the contrapositive of `BitVec.msb_eq_true_iff_large`.
+  TODO: this can be massively cleaned up via `contrapose`, but unfortunately this is in mathli  -/
+theorem msb_eq_false_iff_small {x : BitVec w}
+    : x.msb = false ↔ 2 * x.toNat < 2^w  := by
+  constructor
+  · intros h
+    by_cases hx : (2 * x.toNat ≥ 2^w) <;> try omega
+    exfalso
+    simp_all [msb_eq_true_iff_large (x := x).mpr (by omega)]
+  · intros h
+    rcases hmsb : x.msb with rfl | rfl <;> try rfl
+    exfalso
+    have hcontra := msb_eq_true_iff_large (x := x).mp hmsb
+    omega
+
+/-- Equation theorem for 'Int.sub' when both arguments are 'negSucc' -/
+theorem Int.toNat_sub_toNat_eq_negSucc_ofLt {n m : Nat} (hlt : n < m) :
+  (n : Int) - (m : Int) = (Int.negSucc (m - 1 - n)) := by
+  rw [Int.negSucc_eq] -- TODO: consider adding this to omega cleanup set.
+  omega
```

Comment:
I wonder what the canonical way to achieve this is, I was forced to write a local theorem to make progess in massaging arithmetic.